### PR TITLE
LoHa does not take into account hada_t2 parameter

### PR DESCRIPTION
**Pull request by kovalexal**
Hi!

I've found a strange behaviour, that LoHa doesn't take `hada_t2` parameter into account during training/inference.

I assume that it could be a mistake and almost sure that it should be fixed and it will probably make output results even better.

Could you please confirm that?  If so, we will certainly need to provide some sort of backward compatibility in the following commits.

### CODE REVIEW

**Comment by kovalexal** [comment]
Are you talking about this one https://github.com/7eu7d7/HCP-Diffusion?

**Comment by KohakuBlueleaf** [comment]
[hcp](lycoris/hcp) Folder

**Comment by kovalexal** [comment]
As far as I see, it also has the same [typo](https://github.com/KohakuBlueleaf/LyCORIS/blob/5c967bc4d294d45a77efa25a89a6db6eb885711d/lycoris/hcp/loha.py#L49).

Should it also be fixed there?

**Comment by KohakuBlueleaf** [comment]
Yes

**Comment by kovalexal** [comment]
Ok, I fixed that. But as far as I get it, this modification might brake all existing checkpoints out there.

So we need to come up with some solution for backward compatibility. Maybe we might introduce some flag for which in case it's not present - it will load `t1` weights both to `t1` and `t2`?

**Comment by KohakuBlueleaf** [comment]
> Ok, I fixed that. But as far as I get it, this modification might brake all existing checkpoints out there.
> 
> So we need to come up with some solution for backward compatibility. Maybe we might introduce some flag for which in case it's not present - it will load `t1` weights both to `t1` and `t2`?

I will deal with that
Since this repo is for training only (basically)
So we may able just ignore backward compatibility now

I will check my implementation in sd-webui

**Comment by kovalexal** [comment]
I am trying to adapt your code to be able to use it with Hugging Face PEFT library (PR for [LoHa](https://github.com/huggingface/peft/pull/956), PR for [LoKr](https://github.com/huggingface/peft/pull/978)).

I've tried to infer [this](https://civitai.com/models/27397/loha-renoir-the-dappled-light-style) LoHa checkpoint with [this](https://civitai.com/models/4201/realistic-vision-v51) txt2img model through diffusers and my custom code, and these are the results I get:

```
mode="txt2img",
prompt="Renoir,no human,harbor,scenery",
negative_prompt="duplicate, ugly, huge eyes, ((text)), logo, monochrome, worst face, (bad and mutated hands:1.3), (worst quality:2.0), (low quality:2.0), (blurry:2.0), geometry, (bad hands), (missing fingers), multiple limbs, bad anatomy, (interlocked fingers:1.2), Ugly Fingers, (extra digit and hands and fingers and legs and arms:1.4), (deformed fingers:1.2), (long fingers:1.2), police uniform, police",
seed=42,
scheduler="Euler a",
num_inference_steps=20,
guidance_scale=7,
```

No LoHA:
result_without_loha


LoHa with `hada_t1`, `hada_t1` typo:
result_with_loha_t1_t1


LoHa with `hada_t1`, `hada_t2`:
result_with_loha_t1_t2

I've also dug into the current webui [implementation](https://github.com/AUTOMATIC1111/stable-diffusion-webui/blob/5ef669de080814067961f28357256e8fe27544f4/extensions-builtin/Lora/network_hada.py) and from what I can tell, they are using hada_t1 and hada_t2, which produce different results. I am not quite sure why it still works, probably further normalization neutralizes this effect.

**Comment by KohakuBlueleaf** [comment]
Just a fun fact:
Conv layer normally trained way more less in finetuning a diffusion model

**Comment by kovalexal** [comment]
>Conv layer normally trained way more less in finetuning a diffusion model

Are you talking about learning rate or dropout or other mechanism?

**Comment by KohakuBlueleaf** [comment]
> > Conv layer normally trained way more less in finetuning a diffusion model
> 
> Are you talking about learning rate or dropout or other mechanism?
The diff weight after finetuning.

**Comment by kovalexal** [comment]
@KohakuBlueleaf hi!

Are there any other comments from your side that should be addressed for this PR to get merged into your `dev` branch?

Feel free to share your thoughts in case anything else should be done from my side.

**Comment by KohakuBlueleaf** [comment]
@kovalexal sry I'm too busy recently and forgot this 
Merge now

**Comment by kovalexal** [comment]
@KohakuBlueleaf thank you very much!
